hibernate-core/src/main/java/org/hibernate/engine/internal/Nullability.java

```diff
@@ -95,10 +94,16 @@ public void checkNullability(
 					? persister.getPropertyInsertability()
 					: persister.getPropertyUpdateability();
 			final Type[] propertyTypes = persister.getPropertyTypes();
+			final InMemoryValueGenerationStrategy[] inMemoryValueGenerationStrategies =
```

Comment:
Isn't there the technical possibility to have non-nullable fields with a `InDatabaseValueGenerationStrategy`?

Reply:
Youre right! Didnt think of that :)